### ISSUE TITLE
I've got a permission problem

### ISSUE THREAD

**Issue by ovizii** [meyayl/docker-languagetool#22]
I am using your docker-compose.yml file with slight adaptations according to your instructions here. 

I am facing the following problem:

A docker compose up -d leads to the following error:

`languagetool  | INFO: Changing uid for user "languagetool" to 1000.`

After which the container stops. 
A `docker compose down` followed by `docker compose up -d` repeats the above error.

These are the mapped volumes:

```
    volumes:
      - /opt/languagetool/ngrams:/ngrams
      - /opt/languagetool/fasttext:/fasttext
```

The permissions seem fine:

```
ls -al /opt/languagetool/
total 7
drwxrwxr-x 4 1000 1000   5 Feb 17 18:37 .
drwxr-xr-x 5 root root   5 Jan 27 21:06 .. 
drwxrwxr-x 2 1000 1000   3 Feb 17 18:37 fasttext
drwxrwxr-x 4 1000 1000   4 May  7  2022 ngrams
```

I did try to change the ownership of /opt too, but that didn't change anything:

```
chown 1000:1000 /opt 

ls -al /opt/languagetool/
total 7
drwxrwxr-x 4 1000 1000   5 Feb 17 18:37 .
drwxr-xr-x 5 1000 1000   5 Jan 27 21:06 ..
drwxrwxr-x 2 1000 1000   3 Feb 17 18:37 fasttext
drwxrwxr-x 4 1000 1000   4 May  7  2022 ngrams
```

The only way to get it somehow working is do a `docker compose up -d` followed by a `docker container start languagetool`

```
 docker compose logs -f
languagetool  | INFO: Changing uid for user "languagetool" to 1000.
languagetool  | usermod: Failed to change ownership of the home directoryINFO: Changing uid for user "languagetool" to 1000.
languagetool  | usermod: no changes
languagetool  | INFO: Changing gid for group "languagetool" to 1000.
languagetool  | INFO: Fixing ownership for ngrams base folder if necessary.
languagetool  | INFO: Fixing ownership for fasttext model file if necessasry.
languagetool  | INFO: Skipping download of ngrams model for language en: already exists.
languagetool  | INFO: Skipping download of ngrams model for language de: already exists.
languagetool  | INFO: Skipping download of fasttext model: already exists.
languagetool  | INFO: Creating new LanguageTool config file.
languagetool  | INFO: Version Informations:
languagetool  |   Alpine Linux v3.18.6
languagetool  |   openjdk 17.0.10 2024-01-16
languagetool  |   OpenJDK Runtime Environment Temurin-17.0.10+7 (build 17.0.10+7)
languagetool  |   OpenJDK 64-Bit Server VM Temurin-17.0.10+7 (build 17.0.10+7, mixed mode)
languagetool  | INFO: Using following LanguageTool configuration:
languagetool  |   cacheSize=500
languagetool  |   fasttextBinary=/usr/local/bin/fasttext
languagetool  |   fasttextModel=/fasttext/lid.176.bin
languagetool  |   languageModel=/ngrams
languagetool  |   maxCheckThreads=4
languagetool  |   maxPipelinePoolSize=500
languagetool  |   pipelineCaching=true
languagetool  |   pipelineExpireTimeInSeconds=3600
languagetool  |   pipelinePrewarming=true
languagetool  | INFO: Using JAVA_OPTS=-Xms256m -Xmx3G -XX:+UseSerialGC

```
Any tips on what I am doing wrong here?

**Comment by meyayl** [comment]
The only real error is `usermod: Failed to change ownership of the home directory`, which is irrelevant as it fails to change the home dir of the user, which is not used at all and shouldn't exist in the first place.

I modified the Dockerfile to make sure the user is now created without home directory. Furthmore I made user the uid:gid is only modified if it is not already set.

None of those things explain why your container restarted. I can't explain why it did.

Can you share your complete compose file?

**Comment by ovizii** [comment]
Thanks for having a look. Just to clarify: the container does not restart, it stops after that error. Starting the stopped container again gets it working. 

Here is my compose file. The only thing I was just wondering about, does this container need internet access? I am accessing it behind a reverse proxy (traefik) but the container itself is stuck in an internal network and does not have any routes to the outside. 

`docker exec -ti languagetool bash`

```
route
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
172.16.43.0     *               255.255.255.0   U     0      0        0 eth0
```

```
services:

  languagetool:
    image: meyay/languagetool:latest
    container_name: languagetool
    hostname: languagetool
    restart: "no"
    cap_drop:
      - ALL
    cap_add:
      - CAP_SETUID
      - CAP_SETGID
    security_opt:
      - no-new-privileges
    environment:
      - TZ=Europe/Berlin
      - MAP_UID=1000
      - MAP_GID=1000
      - JAVA_XMS=256m  # OPTIONAL: Setting a minimal Java heap size of 256 mib
      - JAVA_XMX=3G  # OPTIONAL: Setting a maximum Java heap size of 2 Gib
      - download_ngrams_for_langs=en,de
      - langtool_languageModel=/ngrams
      - langtool_fasttextModel=/fasttext/lid.176.bin
      - langtool_pipelinePrewarming=true
      - langtool_pipelineCaching=true
      - langtool_maxPipelinePoolSize=500 
      - langtool_pipelineExpireTimeInSeconds=3600 
      - langtool_cacheSize=500 
      - langtool_maxCheckThreads=4 
    networks:
      - traefik_languagetool
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik_languagetool"
      - "traefik.http.routers.languagetool.tls=true"
      - "traefik.http.routers.languagetool.entrypoints=websecure"
      - "traefik.http.routers.languagetool.rule=Host(`lang.domain.tld`)"
      - "traefik.http.routers.languagetool.middlewares=secHeaders@file"
      - "traefik.http.routers.languagetool.service=languagetool"
      - "traefik.http.services.languagetool.loadbalancer.server.port=8010"
    volumes:
      - /opt/languagetool/ngrams:/ngrams
      - /opt/languagetool/fasttext:/fasttext
      - /dev/shm:/tmp
    cpus: 2
    mem_limit: 4G

networks:

  traefik_languagetool:
      external: true
      internal: true
      name: traefik_languagetool

```

**Comment by meyayl** [comment]
The container requires internet access in case the ngram and fasttext models are not at the expected locations. In total the English model is 15gb of downloads.

**Comment by ovizii** [comment]
My bad, sorry about the confusion, now I understand what was happening. The container immediately stopped because of lack of internet access and that seems to not be checked upon restart hence it worked after starting the stopped container. I just gave it internet access and all works perfectly now. 
I wrongly assumed that it wouldn't need internet access or would tell me so since I had downloaded all necessary ngrams and fasttext.
